docs/chymyst02.md

```diff
@@ -64,20 +64,20 @@ Once the reaction starts, it will receive the value `n = 123` from the input mol
 Both `c(123)` and `f()` will be consumed by the reaction.
 The reaction will then perform the reply action `reply(n)` with `n` set to `123`.
 Only at this point the calling process will get unblocked and receive `123` as the return value of the function call `f()`.
-From the point of view of the reaction that consumed a blocking molecule, the reply action is a non-blocking (i.e. very fast) function call.
+From the point of view of the reaction that consumes a blocking molecule, the reply action is a non-blocking (i.e. very fast) function call.
 The reaction will continue evaluating its reaction body, concurrently with the newly unblocked process that received the reply value `123` and can continue its computations.
 
 This is how the chemical machine implements blocking molecules.
-Blocking molecules work at once as synchronizing barriers and as channels of communication between processes.
+Blocking molecules work at once as [synchronizing barriers](https://en.wikipedia.org/wiki/Barrier_(computer_science)) and as channels of communication between processes.
 
 The syntax for the reply action makes it appear as if the molecule `f` carries _two_ values - its `Unit` value and a special `reply` function, and that the reaction body calls this `reply` function with an integer value.
 However, `f` is injected with the syntax `f()` -- just as any other molecule with `Unit` value.
-The `reply` function appears only in the pattern-matching expression for `f` inside a reaction.
+The `reply` function appears only in the pattern matching expression for `f` inside a reaction.
 
 Blocking molecule injectors are values of type `B[T,R]`, while non-blocking molecule injectors have type `M[T]`.
 Here `T` is the type of value that the molecule carries, and `R` (for blocking molecules) is the type of the reply value.
 
-The pattern-matching expression for a blocking molecule of type `B[T,R]` has the form `case ... + f(v, r) + ...` where `v` is of type `T` and `r` is of type `R => Boolean`.
+The pattern matching expression for a blocking molecule of type `B[T,R]` has the form `case ... + f(v, r) + ...` where `v` is of type `T` and `r` is of type `R => Boolean`.
```

Comment:
Here again "pattern-matching" is used as an adjective, so I think it's easier to read. Otherwise "The pattern maching expression for a molecule" can be misread as "The pattern that matches an expression for a molecule".